packages/firebase_vertexai/firebase_vertexai/lib/src/live_api.dart

```diff
@@ -0,0 +1,302 @@
+          }
+      }; // Or null
+}
+
+/// Represents the available response modalities.
+enum ResponseModalities {
+  /// Unspecified response modality.
+  unspecified('MODALITY_UNSPECIFIED'),
+
+  /// Text response modality.
+  text('TEXT'),
+
+  /// Image response modality.
+  image('IMAGE'),
+
+  /// Audio response modality.
+  audio('AUDIO');
+
+  const ResponseModalities(this._jsonString);
+  final String _jsonString;
+
+  /// Convert to json format
+  String toJson() => _jsonString;
+}
+
+/// Configures live generation settings.
+final class LiveGenerationConfig extends BaseGenerationConfig {
+  // ignore: public_member_api_docs
+  LiveGenerationConfig({
+    this.speechConfig,
+    this.responseModalities,
+    super.candidateCount,
+    super.maxOutputTokens,
+    super.temperature,
+    super.topP,
+    super.topK,
+  });
+
+  /// The speech configuration.
+  final SpeechConfig? speechConfig;
+
+  /// The list of desired response modalities.
+  final List<ResponseModalities>? responseModalities;
+
+  @override
+  Map<String, Object?> toJson() => {
+        ...super.toJson(),
+        if (speechConfig case final speechConfig?)
+          'speechConfig': speechConfig.toJson(),
+        if (responseModalities case final responseModalities?)
+          'responseModalities':
+              responseModalities.map((modality) => modality.toJson()).toList(),
+      };
+}
+
+/// An abstract class representing a message received from a live server.
+///
+/// This class serves as a base for different types of server messages,
+/// such as content updates, tool calls, and tool call cancellations.
+/// Subclasses should implement specific message types.
+abstract class LiveServerMessage {}
+
+/// A message indicating that the live server setup is complete.
+///
+/// This message signals that the initial connection and setup process
+/// with the live server has finished successfully.
+class LiveServerSetupComplete implements LiveServerMessage {}
+
+/// Represents content generated by the model in a live stream.
+class LiveServerContent implements LiveServerMessage {
+  /// Creates a [LiveServerContent] instance.
+  ///
+  /// [modelTurn] (optional): The content generated by the model.
+  /// [turnComplete] (optional): Indicates if the turn is complete.
+  /// [interrupted] (optional): Indicates if the generation was interrupted.
+  LiveServerContent({this.modelTurn, this.turnComplete, this.interrupted});
+
+  /// The content generated by the model.
+  final Content? modelTurn;
+
+  /// Indicates if the turn is complete.
+  final bool? turnComplete;
+
+  /// Indicates if the generation was interrupted.
+  final bool? interrupted;
+}
+
+/// Represents a tool call in a live stream.
+class LiveServerToolCall implements LiveServerMessage {
+  /// Creates a [LiveServerToolCall] instance.
+  ///
+  /// [functionCalls] (optional): The list of function calls.
+  LiveServerToolCall({this.functionCalls});
+
+  /// The list of function calls.
```

Comment:
Can you elaborate? Make a distinction whether this is the a list of functions that _can be_ called, or if it's a list of functions that _have been_ called.

Reply:
done.